elastic/kibana
# [Upgrade Assistant] New Kibana deprecations page

This PR updates the Kibana deprecations page in Upgrade Assistant to align with the new design prototype.

### How to test
There should be a number of config deprecations that display out of the box. To add a deprecation that supports the "Quick resolve" functionality, follow these steps:
1. Create a PDF report (you can easily do this from Dashboard using the sample data)
2. Issue the following curl command as outlined in https://github.com/elastic/kibana/pull/104303:
```
curl --request PUT \
  --url 'http://localhost:9200/.reporting-*/_settings' \
  --header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==' \
  --header 'Content-Type: application/json' \
  --data '{
	"settings": {
		"index.lifecycle.name": null
	}
}'
```

### Deferred work
There are some features that the core team is looking into supporting within the Kibana deprecations service. If/when these are implemented, I will create a follow-up PR for UA.
1. Markdown support for manual steps
2. Requiring the "deprecationType" field when registered a deprecation. This is currently optional, and UA defaults to "uncategorized" when not specified.
3. ~Requiring a deprecation has a title (https://github.com/elastic/kibana/pull/109840). This will allow us to remove the hard-coded title, "XX is using a deprecated feature".~ Added support via https://github.com/elastic/kibana/pull/110101/commits/677363646bb9dc2403af31a07129168d7feb40ef

### Screenshots

<details>
<summary>Overview of changes</summary>

<img width="1166" alt="Screen Shot 2021-08-31 at 1 20 18 PM" src="https://user-images.githubusercontent.com/5226211/131547970-c57cae3a-7a26-4552-891e-4665105a8a2e.png">

<img width="1168" alt="Screen Shot 2021-08-31 at 1 20 57 PM" src="https://user-images.githubusercontent.com/5226211/131548225-9384e78b-a875-4ce3-8c09-6f4ce15877b2.png">
<img width="1180" alt="Screen Shot 2021-08-31 at 1 21 15 PM" src="https://user-images.githubusercontent.com/5226211/131548229-56dfdb00-8251-4834-9d3a-b83a13d36d11.png">
<img width="1644" alt="Screen Shot 2021-08-31 at 1 25 43 PM" src="https://user-images.githubusercontent.com/5226211/131548364-4e1eef85-653c-4085-b6a6-e10c0a052024.png">
<img width="1640" alt="Screen Shot 2021-08-31 at 1 31 45 PM" src="https://user-images.githubusercontent.com/5226211/131549191-d592804b-3433-4bac-b3b6-4fb9b8757f2b.png">

// Quick resolve flyout
<img width="1097" alt="Screen Shot 2021-08-31 at 1 17 42 PM" src="https://user-images.githubusercontent.com/5226211/131548416-1fb69edf-3aa8-4939-9981-b18c0452311f.png">

// Manual steps only flyout
<img width="1652" alt="Screen Shot 2021-08-31 at 1 26 51 PM" src="https://user-images.githubusercontent.com/5226211/131548505-f223a440-ca6c-4886-ba83-b5614f1a9fb0.png">


// Deprecation that has been resolved
<img width="1101" alt="Screen Shot 2021-08-31 at 1 17 50 PM" src="https://user-images.githubusercontent.com/5226211/131548042-964deadf-b8fa-4832-b94f-f4140a4f0ff1.png">

<img width="1106" alt="Screen Shot 2021-08-31 at 1 17 58 PM" src="https://user-images.githubusercontent.com/5226211/131548103-2b6c1138-e791-422f-af9f-2031cc343440.png">

// Deprecation with failure
<img width="1087" alt="Screen Shot 2021-08-31 at 1 16 50 PM" src="https://user-images.githubusercontent.com/5226211/131548071-e7804a8e-d15e-4488-aacc-0a300c79a6c4.png">

<img width="1126" alt="Screen Shot 2021-08-31 at 1 16 43 PM" src="https://user-images.githubusercontent.com/5226211/131548158-bff9ae05-6d13-40f6-ab06-e469e044dd7d.png">

</details>
